### PR TITLE
Task7

### PR DESCRIPTION
**Pull request by arkud**
Done.

### CODE REVIEW

File: task7/script.js [hunk 1]
```diff
@@ -0,0 +1,369 @@
...
+            if (post.name === undefined) {
+                console.log("wrong name\n");
+                return false;
+            }
+            return true;
+        }
+
+        static updateHashtags(post) {
+            let hashtagRegExp = /#\S+/;
+            post.hashtags = hashtagRegExp.exec(post.description);
```
Comment (ghost) [inline task7/script.js]: Incorrect, exec find fist matching and returns array of regexp group
e.g. /#(\s+)/.exec('Hi #hash tag #example')  // ['#hash', 'hash'] - where [full match, 1 group]

File: task7/script.js [hunk 2]
```diff
@@ -0,0 +1,369 @@
...
+            skip = skip || 0;
+            top = top || 10;
+
+            let result = this._posts.slice();
+
+            if (filterConfig.hashtags) {
+                result = result.filter(function (item) {
+                    if (item.hashtags === null || item.hashtags === undefined)
+                        return false;
+                    for (let i = 0; i < filterConfig.hashtags.length; i++) {
```
Comment (ghost) [inline task7/script.js]: Array.prototype.some

File: task7/script.js [hunk 3]
```diff
@@ -0,0 +1,369 @@
...
+
+        getPosts(skip, top, filterConfig) {
+            skip = skip || 0;
+            top = top || 10;
+
+            let result = this._posts.slice();
+
+            if (filterConfig.hashtags) {
+                result = result.filter(function (item) {
+                    if (item.hashtags === null || item.hashtags === undefined)
```
Comment (ghost) [inline task7/script.js]: It's better to check direct type
e.g.  (Array.isArray(item.hashtags)),  (item.date instanceof Date)

File: task7/script.js [hunk 4]
```diff
@@ -0,0 +1,369 @@
...
+                        return false;
+                    for (let i = 0; i < filterConfig.hashtags.length; i++) {
+                        if (item.hashtags.includes(filterConfig.hashtags[i]))
+                            return true;
+                    }
+                });
+            }
+
```
Comment (ghost) [inline task7/script.js:249]: filtering but not sorting by date / name ?

File: task7/script.js [hunk 5]
```diff
@@ -0,0 +1,369 @@
...
+            return false;
+        }
+
+        editPost(id, post) {
+            console.log("Change post with id:" + id);
+            if (post.createdAt || post.author || post.name || post.id) {
+                console.log("restricted fields cannot be changed");
+                return false;
+            }
+            if (!ChirikList.validatePost(this.getPost(id)))
```
Comment (ghost) [inline task7/script.js]: Does the stored post can be invalid ?

**Review by ghost — APPROVED** [review]
Approved for 6.5

File: task7/script.js (hunk 4 above)
Comment (arkud) [reply task7/script.js:249]: Code for that is just below.

File: task7/script.js (hunk 2 above)
Comment (arkud) [reply task7/script.js]: Done.

File: task7/script.js (hunk 3 above)
Comment (arkud) [reply task7/script.js]: Done.

File: task7/script.js (hunk 4 above)
Comment (arkud) [reply task7/script.js:249]: Code for that is just below.

File: task7/script.js (hunk 5 above)
Comment (arkud) [reply task7/script.js]: Removed this if.

File: task7/script.js (hunk 1 above)
Comment (arkud) [reply task7/script.js]: Now works properly.

File: task7/script.js (hunk 4 above)
Comment (ghost) [reply task7/script.js:249]: Filtering is not sorting !

**Comment by ghost** [comment]
Mark updated.

File: task7/script.js (hunk 4 above)
Comment (arkud) [reply task7/script.js:249]: The lines 250-254 sort posts, after the filter by hashtags.

Comment (ghost) [reply task7/script.js:249]: According to TP getPage should have filtering by hashtags, author and day + multiple filters support.
Sorting should be by date - the latest tweet should be first. 
E.g. :
ChirikList.getPage(0, 10, {name: 'ar_kud', before: new Date('27.05.2020'), hashtag: ['sometag']}) // [{...}, {...}]

**Pull request merged**